### DISCUSSION TITLE
How to streamline the UX for newcomers

### DISCUSSION THREAD

**Discussion by grayswandyr** [ocaml/dune#12739] [General]
Following discussions on the OCaml forum, I'd like to advocate the implementation of features that would, in my mind, simplify the whole experience for newcomers and developers of simple projects (for a definition of simple that dune developers can refine). As a professor, we often meet the following issues with our students:

* 3 files, and even 4 with .ocamlformat, are necessary just to build/run/test a few OCaml files (I'm not asking for switching off from sexps but this issue combines with the facts that sexps are unknown to our students, so they also don't know what to type and in which file)
* getting a combination of tests and non-fatal warnings (very useful when teaching) requires a stanza that no one can remember (besides, I seem to recall that unaddressed warnings are not displayed again when rebuilding, which is not cool in this configuration)

@rgrinberg had sketched the idea of possibly having a single file (dune-project IIRC) for simple projects. In the context of the OCaml platform, and comparing to other famous recent platforms, I'd like to advocate that this should be a priority, together with a good warning and test story (meaning such important activities as test and repeated non-fatal warning should be easily activated, not through a complex stanza).

**Comment by smolkaj** [comment]
I have similar concerns, explained  in a footnote of [this comment](https://discuss.ocaml.org/t/in-praise-of-ocaml/13567/2?u=smolkaj) (find and click the three dots to see it) and elaborated in [this comment](https://discuss.ocaml.org/t/in-praise-of-ocaml/13567/4?u=smolkaj).

**Comment by shonfeder** [comment]
I think the goal of this issues is important, but it is more of a general design principle than an actionable issues.

IIUC, there are two concrete proposals here, and they are tracked in other issues:

1. A no-configuration/low configuration build mode, tracked in https://github.com/ocaml/dune/issues/1385
2. More ergonomic warning/error configuration, tracked in https://github.com/ocaml/dune/issues/1933

However, I think this is an important topic for discussion. So I am going to move this into the discussions.
